packages/expo/src/Google/Google.ts

```diff
@@ -25,53 +41,123 @@ export type LogInResult =
     }
   | {
       type: 'success';
-      accessToken?: string;
+      accessToken: string | null;
       idToken: string | null;
       refreshToken: string | null;
       user: GoogleUser;
     };
 
-export async function logInAsync(config: LogInConfig): Promise<LogInResult> {
-  if (!AppAuth.authAsync) {
-    throw new UnavailabilityError('AppAuth', 'logInAsync');
+function getPlatformGUID(config: LogInConfig) {
+  const { clientId } = config;
+
+  const iosClientId =
+    Constants.appOwnership === 'standalone' ? config.iosClientId : config.iosStandaloneAppClientId;
+  const androidClientId = isInExpo ? config.androidClientId : config.androidStandaloneAppClientId;
+
+  const platformClientId =
+    Platform.select({
+      ios: iosClientId,
+      android: androidClientId,
+      default: config.clientId,
+    }) || clientId;
+
+  if (
+    typeof iosClientId === 'string' &&
+    typeof androidClientId === 'string' &&
+    iosClientId === androidClientId
+  ) {
+    throw new CodedError(
+      'ERR_GOOGLE_CONFIG',
+      'Keys for Android an iOS cannot be the same value. Ensure you are linking the client IDs matching the given platforms in the Google APIs console: https://console.developers.google.com/apis/credentials'
+    );
+  }
+
+  const guid = guidFromClientId(platformClientId);
+  return guid;
+}
+
+// TODO: Bacon: ensure this is valid for all cases.
+const PROJECT_NUMBER_LENGTH = 11;
+
+const PROJECT_ID_LENGTH = 32;
+
+function isValidGUID(guid: string) {
+  const components = guid.split('-');
+  if (components.length !== 2) {
+    return {
+      isValid: false,
+      reason: `\`${guid}\` must be a string of numbers and an alphanumeric string ${PROJECT_ID_LENGTH} characters long, joined with a hyphen.`,
+    };
+  }
+  const projectNumber = components[0];
+  const projectId = components[1];
+  if (isNaN(+projectNumber)) {
+    const hashedProjectId = Array(PROJECT_ID_LENGTH).map((l, i) => (i % 2 === 0 ? 'X' : 'x'));
+    return {
+      isValid: false,
+      reason: `\`${projectNumber}-${hashedProjectId}\` project number must be a string of numbers.`,
+    };
+  }
+  if (!projectId.match('^[a-zA-Z0-9]+$')) {
+    const hashedProjectNumber = Array(projectNumber.length).map((l, i) =>
+      i % 2 === 0 ? 'X' : 'x'
+    );
+    return {
+      isValid: false,
+      reason: `\`${hashedProjectNumber}-${projectId}\` project ID must be an alphanumeric string ${PROJECT_ID_LENGTH} characters long.`,
+    };
+  }
+
+  return { isValid: true };
+}
+
+function guidFromClientId(clientId: string): string {
+  const clientIdComponents = clientId.split('.').filter(component => component.includes('-'));
+
+  const guid = clientIdComponents[0];
+  const isValid = isValidGUID(guid);
+  if (!isValid.isValid) {
```

Comment:
lol name this a bit differently:
```suggestion
  if (!validation.isValid) {
```

Reply:
woopsies